File: p097.java
/* 
 * Solution to Project Euler problem 97
 * By Nayuki Minase
 */

import java.math.BigInteger;


public class p097 {
	
	public static void main(String[] args) {
		BigInteger modulus = BigInteger.TEN.pow(10);
		BigInteger n = BigInteger.valueOf(2).modPow(BigInteger.valueOf(7830457), modulus);
		n = n.multiply(BigInteger.valueOf(28433)).mod(modulus);
		n = n.add(BigInteger.ONE).mod(modulus);
		System.out.printf("%010d%n", n);
	}
	
}

File: p097.hs
{- 
 - Solution to Project Euler problem 97
 - By Nayuki Minase
 - 
 - http://nayuki.eigenstate.org/page/project-euler-solutions
 - https://github.com/nayuki/Project-Euler-solutions
 -}


main = putStrLn (show ans)
ans = mod (28433 * 2^7830457 + 1) (10^10)
